### ISSUE TITLE
Multiple issues (Edit: All fixed except freezing issue)

### ISSUE THREAD

**Issue by CecilFF4** [nicoboss/nsz#51]
Thank you for updating this project. There are a few reasons I'll be staying on 2.1.1, however.

~~3.1.1 won't work without the master_key_07. Not even on games/dlc that don't use that. I also acknowledge that this issue may cause some other ones listed here.~~

~~Freezes on doing any action (completion bar stays at 0% or 100%).~~

The GUI shuts down while the command prompt stays up.

GUI doesn't allow drag-and-drop.

GUI doesn't open a Windows-style prompt for directory/file information. More hassle to use 3.1.1 than 2.1.1 (I have a batch file for 2.1.1 with the directories already set, so I don't have to click through a dozen folders each time).

OS: Win10 x64

EDIT: Very sorry about wasting your time. I dumped my keys using the proper payload but didn't copy them from the correct folder (copied an old keys file from a different place). Keys fixed, program appears to work with files it didn't before.

**Comment by nicoboss** [comment]
Thanks for your feedback.
- I will look into the masterkey issue. On versions prior to 3.0.0 missing keys warnings were just not displayed so it shouldn't matter but further testing is required.
- I will look into the freezing thing too. Can you please let me know for what game it occurs?
- You can still use command line control on 3.1.1. The GUI is just an additional way to use nsz for less experienced users.
- The GUI is meant to shut down after it sent the action to the actual tool as any further information will be displayed inside the command line prompt.
- Reason for this customized file/folder selector is Linux and Mac OS X support which doesn't have the windows folder browser and using that would break cross platform support
- Drag and drop is also very OS specific and not really supported by Kivy but I look into it if there is any way to add that but I don't think there is.

**Comment by CookieXD** [comment]
Also got the following error:
```
Traceback (most recent call last):
  File "nsz\nut\Keys.py", line 107, in load
OSError: master_key_08 missing from keys.txt
```

But the thing is, on my Computer (Windows) is no file called keys.txt
I seached the whole drive.
Where is that file supposed to be?

**Comment by CecilFF4** [comment]
The freezing occurred for most NSPs I tried tossing into it. There are some that just refused to work, regardless of whether the compression was attempted on 2.1.1 or 3.1.1 (list below). Others would seem like they compressed but stay at 100% indefinitely. The files may have finished, but the command prompt didn't auto-close or say it was done. I don't have examples of those since the NSPs that I compressed successfully using 2.1.1 were deleted.

> Cadence of Hyrule
> Cadence of Hyrule (update)
> Cuphead
> Darkwood
> Darkwood (update)
> Final Fantasy 9 (update)
> Katana ZERO (update)
> Out There Omega The Alliance
> Super Mario Party (update)
> Legend of Zelda Link's Awakening (update)
> When Ski Lifts Go Wrong (update)
> Yoshi's Crafted World (update)

**Comment by nicoboss** [comment]
All your issues seem to be fixed with latest NSZ v4.0 release please reopen if you can still recreate any of them with that version.
